scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -383,19 +383,41 @@ class FossId(
         return results
     }
 
+    private suspend fun List<Scan>.findLatestPendingOrFinishedScan(url: String, revision: String? = null): Scan? {
```

Comment:
One more minor nit: As the function now consists of a single expression, you can drop the curly braces and the return statement.